### PR TITLE
Feat/share service route and page

### PR DESCRIPTION
**Pull request by 0xLodz**
This PR introduces route and initialisation wrapper for service sharing, enabling access to share service page. 

### Summary by CodeRabbit

**New Feature:**
- Added a new `ShareService` page that includes a step-by-step process for service reference selection. This feature enhances the user interface by providing a more structured and intuitive way to share services.
- Updated the `Share` page with improved interactivity. The hover effect on the `.right` class has been removed, and a click event has been added to navigate to the `ShareService` page.
- Implemented internationalization for the title of the `ShareService` page, improving accessibility for users in different locales.

**Refactor:**
- Introduced a new route `/share/service` for the `ShareService` page, improving the organization and readability of the application's routing structure.

### CODE REVIEW

File: src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@import '@/ui/style/fonts.scss';
+@import '@/ui/style/theme.scss';
+@import '@/ui/style/mixins.scss';
```
Comment (lolottetheclash) [inline src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.scss:3]: same here

File: src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import type { FC } from 'react'
+import './serviceReferenceSelection.scss'
+
+export const ServiceReferenceSelection: FC = () => (
+  <div className="okp4-dataverse-portal-share-service-page-service-reference-container">
+    <h2 className="okp4-dataverse-portal-share-service-page-service-reference-title"></h2>
```
Comment (lolottetheclash) [inline src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.tsx:6]: Maybe you can just add the title content

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
Thanks for this new step! tiny comments before merging (we'll need to think about sharing the code between share dataset and share service )

**Review by fredericvilcot — APPROVED** [review]
👍

File: src/ui/page/share/service/shareService.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
+@import '@/ui/style/fonts.scss';
+@import '@/ui/style/theme.scss';
+@import '@/ui/style/mixins';
```
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/page/share/service/shareService.scss:3]: oops, thanks!

File: src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.scss (hunk 1 above)
Comment (0xLodz) [reply src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.scss:3]: oops, thanks!

File: src/ui/page/share/service/shareService.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/page/share/service/shareService.scss:3]: We need it for the `norse-gradient-header` mixin :)

File: src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import type { FC } from 'react'
+import './serviceReferenceSelection.scss'
+
+export const ServiceReferenceSelection: FC = () => (
+  <div className="okp4-dataverse-portal-share-service-page-service-reference-container">
+    <h2 className="okp4-dataverse-portal-share-service-page-service-reference-title">
+      Content title
+    </h2>
+  </div>
+)
```
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/page/share/service/steps/serviceReferenceSelection/serviceReferenceSelection.tsx:10]: @coderabbitai this is a temporary title. Thanks ;)

**Review by lolottetheclash — APPROVED** [review]

**Pull request merged**